---
title: "Tables"
author: "Risha Govind"
date: "`r Sys.Date()`"
output: word_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning=FALSE, size="smallsize")
library(lazyeval)
library(dplyr)
library(pander)
library(gdata)
require(gdata)

```


```{r load data}
## Load datasets

#BRUID to Spanish IDS [From Rachel]
BRUID_localID <- read.csv("../BRUID_localID.csv", header=T, sep=',') %>%
  tbl_df

#List of Samples from ACM TTNtv Group [From Liz]
ACM_TTNtv_Group <- read.table("../ACM_TTNtv_group.txt", header=T) %>%
  tbl_df

#List of Samples from ACM DCM_var_nonTTN Group [From Liz]
ACM_DCM_var_nonTTN_Group <- read.table("../ACM_DCM_var_nonTTN_group.txt", header=T) %>%
  tbl_df

#Clinical Phenotype of ACM Samples [From Spanish Team]
ACM_pheno = read.xls("../ACM_pheno.xlsx", sheet=1, header=T, fileEncoding="latin1") 

#English translations of headers of ACM_pheno file [From Spanish Team]
ACM_english_headers = read.csv ("../ACM_ColumnNames_English.csv", header = FALSE, sep=',', fileEncoding="latin1")

#dim(ACM_pheno)
```


```{r removing empty rows and columns from excel file dataframe}
## removing blank rows/columns from excel file dataframe

#remove blank columns
ACM_pheno <- ACM_pheno[,colSums(is.na(ACM_pheno))<nrow(ACM_pheno)]
#remove rows that do not have sampleIDs
ACM_pheno <- ACM_pheno[!(ACM_pheno$SAMPLES  == ""), ]  %>%
  tbl_df


#dim(ACM_pheno)

```

 
```{r update spanish col names to english in ACM_pheno}

## Update Headers to English

ACM_pheno_colNames = colnames(ACM_pheno)

count <- nrow(ACM_english_headers)

#replace all spaces and dots with underscore
ACM_pheno_colNames<- gsub(x = ACM_pheno_colNames,pattern = " ", replacement = "_",fixed = TRUE)
ACM_pheno_colNames<- gsub(x = ACM_pheno_colNames,pattern = ".", replacement = "_",fixed = TRUE)

for (i in 1:count){
  headerSpanish <- ACM_english_headers$V1[i]
  headerEnglish <- ACM_english_headers$V2[i]
  
  #replace all spaces and dots with underscore
  headerSpanish<- gsub(x = headerSpanish,pattern = " ", replacement = "_",fixed = TRUE)
  headerEnglish<- gsub(x = headerEnglish,pattern = " ", replacement = "_",fixed = TRUE)
  headerSpanish<- gsub(x = headerSpanish,pattern = ".", replacement = "_",fixed = TRUE)
  headerEnglish<- gsub(x = headerEnglish,pattern = ".", replacement = "_",fixed = TRUE)


  ACM_pheno_colNames<-gsub(headerSpanish, headerEnglish, ACM_pheno_colNames,ignore.case = TRUE)  
}

#manual assignments, not in the coversion table or were not comparable due to typos
ACM_pheno_colNames<-gsub("FibrilaAuricular", "ATRIAL_FIBRILLATION", ACM_pheno_colNames,ignore.case = TRUE)
ACM_pheno_colNames<-gsub("Fecha.DEATH", "DATE_OF_DEATH", ACM_pheno_colNames,ignore.case = TRUE)
ACM_pheno_colNames<-gsub("Fecha.TRANSPLANT","DATE_OF_TRANSPLANT", ACM_pheno_colNames,ignore.case = TRUE)
ACM_pheno_colNames<-gsub("EDAD.UltC","AGE_AT_LAST_CONSULTATION", ACM_pheno_colNames,ignore.case = TRUE)
ACM_pheno_colNames<-gsub("Causa_MUER","CAUSE_OF_DEATH", ACM_pheno_colNames,ignore.case = TRUE)
ACM_pheno_colNames<-gsub("FECHA_NACMIENTO","BIRTHDATE", ACM_pheno_colNames,ignore.case = TRUE)
ACM_pheno_colNames<-gsub("RITHM.","RHYTHM", ACM_pheno_colNames,ignore.case = TRUE)
ACM_pheno_colNames<-gsub("BETABLOQ","BETABLOCKER", ACM_pheno_colNames,ignore.case = TRUE)
ACM_pheno_colNames<-gsub("BebeActual","CURRENT_DRINKER", ACM_pheno_colNames,ignore.case = TRUE)
ACM_pheno_colNames<-gsub("INITIAL_EYECTION_FRACTION","INITIAL_EJECTION_FRACTION", ACM_pheno_colNames,ignore.case = TRUE)
ACM_pheno_colNames<-gsub("DATE_ONSET_OF_SYNTOMS","DATE_ONSET_OF_SYMPTOMS", ACM_pheno_colNames,ignore.case = TRUE)
ACM_pheno_colNames<-gsub("AGE_AT_INICIAL_CLINICAL_ASSESSMENT","AGE_AT_INITIAL_CLINICAL_ASSESSMENT", ACM_pheno_colNames,ignore.case = TRUE)
ACM_pheno_colNames<-gsub("HISTORIA_FAMILIAR_DE_DEATH_SUBITA","FAMILY_HISTORY_OF_SUDDEN_CARDIAC_DEATH", ACM_pheno_colNames,ignore.case = TRUE)
ACM_pheno_colNames<-gsub("RECUPERA_FEVI","EJECTION_FRACTION_RECOVERY", ACM_pheno_colNames,ignore.case = TRUE)
ACM_pheno_colNames<-gsub("QRS_DURATION_120","QRS_gt_120", ACM_pheno_colNames,ignore.case = TRUE)


#add column names to ACM_pheno
colnames(ACM_pheno)<- ACM_pheno_colNames


```

```{r tidy datasets to prepare for merge}
#remove irrelevant columns 
BRUID_localID <-BRUID_localID[,c("LocalIdNumber","BruNumber")]

#update the LocalIdNumber of one sample was does not match with the pheno file 
BRUID_localID$LocalIdNumber <- as.character(BRUID_localID$LocalIdNumber)
BRUID_localID$LocalIdNumber[BRUID_localID$LocalIdNumber == 'BELL-508 BIS'] <- "BELL-508"

#Add BRU IDs to pheno table
ACM_pheno_BRUID<-merge(BRUID_localID, ACM_pheno, by.x = "LocalIdNumber", by.y = "SAMPLES", all = TRUE) 

#Tidy dataframes of ACM groups [TTNtv, DCMvar_nonTTN, GenoNeg]

#select first column only
ACM_TTNtv_Group<-ACM_TTNtv_Group[,1]
#rename first column to BRUID
colnames(ACM_TTNtv_Group) <- "BRUID"
#Add a second column : VarGroup
ACM_TTNtv_Group<-cbind(ACM_TTNtv_Group,VarGroup="TTNtv")

#select first column only
ACM_DCM_var_nonTTN_Group<-ACM_DCM_var_nonTTN_Group[,1]
#rename first column to BRUID
colnames(ACM_DCM_var_nonTTN_Group) <- "BRUID"
#Add a second column : VarGroup
ACM_DCM_var_nonTTN_Group<-cbind(ACM_DCM_var_nonTTN_Group,VarGroup="DCMVariant_nonTTN")

#Remove TTNtv samples from ACM_DCM_var_nonTTN_Group this. 20PC04885 has a TTNtv and a LMNA SNP. This sample is to be in the TTNtv group 
ACM_DCM_var_nonTTN_Group<-subset(ACM_DCM_var_nonTTN_Group, BRUID!="20PC04885")

ACM_groups=rbind(ACM_TTNtv_Group,ACM_DCM_var_nonTTN_Group)

```



```{r combine data}

## Combine all datasets
ACM <- merge(ACM_groups, ACM_pheno_BRUID, by.x = "BRUID", by.y = "BruNumber", all = TRUE)

#Update VarGroup of GenotypeNegative
ACM$VarGroup <- as.character(ACM$VarGroup)
ACM$VarGroup[is.na(ACM$VarGroup)] <- "GenotypeNegative"

#dim(ACM)

```

```{r Prepare data for analysis}
#SEX is coded 1=Male, 2=Female, change 0=Female
#For easier calculations, it's easier to work with 1 and 0
ACM$SEX[ACM$SEX=="2"] <- 0
```


```{r Get Mean and Std.Dev}

#To Do
# AGE oF SYMPTOM

continuousVariable_summary <- function(x){
  c<-paste(round(mean(x,na.rm=T), digits=2)," \u00b1 ", round(sd(x,na.rm=T), digits=2), sep="")
return(c)
}

binaryVariable_summary <- function(e,f){
  b<-paste(round(sum(e,na.rm=T))," \u0028",round(((sum(e,na.rm=T)/f) * 100), digits=1),"\u0025\u0029", sep="")
return(b)
}


#To Do
# AGE oF SYMPTOM
# NYHA CLASS 
# Did `na.rm=T` here, because there was some missing data. Need to check if this appropriate.

MainTable1 <- ACM %>%
group_by(VarGroup) %>%
  summarise(
    #total=n(),
    #continuous Variables
    Gr_alcohol=continuousVariable_summary(Gr_alcohol),
    AGE_AT_INITIAL_CLINICAL_ASSESSMENT=continuousVariable_summary(AGE_AT_INITIAL_CLINICAL_ASSESSMENT),
    INITIAL_EJECTION_FRACTION=continuousVariable_summary(INITIAL_EJECTION_FRACTION),
    DIASTOLIC_DIAMETER=continuousVariable_summary(DIASTOLIC_DIAMETER),
    #binary Variables
    SEX=binaryVariable_summary(SEX,n()),
    ATRIAL_FIBRILLATION=binaryVariable_summary(ATRIAL_FIBRILLATION,n()),
    FAMILY_HISTORY_OF_CARDIOMYOPATHY=binaryVariable_summary(FAMILY_HISTORY_OF_CARDIOMYOPATHY,n()),
    FAMILY_HISTORY_OF_SUDDEN_CARDIAC_DEATH=binaryVariable_summary(FAMILY_HISTORY_OF_SUDDEN_CARDIAC_DEATH,n())
    ) 

```



```{r get number of NAs}
MainTable2 <- ACM %>%
  group_by(VarGroup) %>%
  summarise(
    Gr_alcohol=sum(is.na(Gr_alcohol)),
    AGE_AT_INITIAL_CLINICAL_ASSESSMENT=sum(is.na(AGE_AT_INITIAL_CLINICAL_ASSESSMENT)),
    INITIAL_EJECTION_FRACTION=sum(is.na(INITIAL_EJECTION_FRACTION)),
    DIASTOLIC_DIAMETER=sum(is.na(DIASTOLIC_DIAMETER)),
    SEX=sum(is.na(SEX)),
    ATRIAL_FIBRILLATION=sum(is.na(ATRIAL_FIBRILLATION)),
    FAMILY_HISTORY_OF_CARDIOMYOPATHY=sum(is.na(FAMILY_HISTORY_OF_CARDIOMYOPATHY)),
    FAMILY_HISTORY_OF_SUDDEN_CARDIAC_DEATH=sum(is.na(FAMILY_HISTORY_OF_SUDDEN_CARDIAC_DEATH))
  ) 
  
```

```{r get P-values}
#Mann-Whitney or wilcox.test : continuous (not normal) vs categorical (binary) 
#Fisher Test : categorical (binary) vs categorical (binary) 


MainTable3 <- ACM %>%
  filter(VarGroup=="TTNtv" | VarGroup=="GenotypeNegative") %>%
  summarise(
    Gr_alcohol=with(.,round(digits=4,wilcox.test(Gr_alcohol ~ VarGroup)$p.val)),
    AGE_AT_INITIAL_CLINICAL_ASSESSMENT=with(.,round(digits=4,wilcox.test(AGE_AT_INITIAL_CLINICAL_ASSESSMENT ~ VarGroup)$p.val)),
    INITIAL_EJECTION_FRACTION=with(.,round(digits=4,wilcox.test(INITIAL_EJECTION_FRACTION ~ VarGroup)$p.val)),
    DIASTOLIC_DIAMETER=with(.,round(digits=4,wilcox.test(DIASTOLIC_DIAMETER ~ VarGroup)$p.val)),
    SEX=with(., round(digits=4,fisher.test(x=SEX,y=VarGroup)$p.val)),
    ATRIAL_FIBRILLATION=with(., round(digits=4,fisher.test(x=ATRIAL_FIBRILLATION,y=VarGroup)$p.val)),
    FAMILY_HISTORY_OF_CARDIOMYOPATHY=with(., round(digits=4,fisher.test(x=FAMILY_HISTORY_OF_CARDIOMYOPATHY,y=VarGroup)$p.val)),
    FAMILY_HISTORY_OF_SUDDEN_CARDIAC_DEATH=with(., round(digits=4,fisher.test(x=FAMILY_HISTORY_OF_SUDDEN_CARDIAC_DEATH,y=VarGroup)$p.val))
  )
```

## Table 1 - Within ACM Cohort

```{r Combine all above :Table 1}
#To DO
# Number in each group in column names is hard coded. Need to be fixed  

MainTable1<-MainTable1[,c("Gr_alcohol","AGE_AT_INITIAL_CLINICAL_ASSESSMENT","INITIAL_EJECTION_FRACTION","DIASTOLIC_DIAMETER","SEX","ATRIAL_FIBRILLATION","FAMILY_HISTORY_OF_CARDIOMYOPATHY","FAMILY_HISTORY_OF_SUDDEN_CARDIAC_DEATH")]
MainTable2<-MainTable2[,c("Gr_alcohol","AGE_AT_INITIAL_CLINICAL_ASSESSMENT","INITIAL_EJECTION_FRACTION","DIASTOLIC_DIAMETER","SEX","ATRIAL_FIBRILLATION","FAMILY_HISTORY_OF_CARDIOMYOPATHY","FAMILY_HISTORY_OF_SUDDEN_CARDIAC_DEATH")]
MainTable3<-MainTable3[,c("Gr_alcohol","AGE_AT_INITIAL_CLINICAL_ASSESSMENT","INITIAL_EJECTION_FRACTION","DIASTOLIC_DIAMETER","SEX","ATRIAL_FIBRILLATION","FAMILY_HISTORY_OF_CARDIOMYOPATHY","FAMILY_HISTORY_OF_SUDDEN_CARDIAC_DEATH")]

MainTable<-rbind(MainTable1,MainTable2,MainTable3)
MainTable<-t(MainTable)

#change order of columns
Table1<-MainTable[,c("3","2","1","7")]

colnames(Table1) <- c("TTNtv<br> (n=14)","Genotype Neg <br> (n=122)","Other Variants <br> (n=5)","P-value <br> (TTNtv vs Genotype Neg)")

rownames(Table1) <- c("Alcohol (grams)","Age at initial clinical assesement","Initial ejection fraction" ,"Diastolic diameter","Sex (males)","Atrial fibrillation","Family history of CM","Family history of SCD")

                         
panderOptions('table.split.table', 800)
pander(Table1)

```

## Table for internal use - missing data info for Table 1
```{r}
Table1_extra<-MainTable[,c("6","5","4")]

colnames(Table1_extra) <- c("TTNtv <br> (NA's)","Genotype Neg <br> (NA's)","Other Variants <br> (NA's)")

rownames(Table1_extra) <- c("Alcohol (grams)","Age at initial clinical assesement","Initial ejection fraction" ,"Diastolic diameter","Sex (males)","Atrial fibrillation","Family history of CM","Family history of SCD")

panderOptions('table.split.table', 800)
pander(Table1_extra)

```

## Supplementary Table 1 
```{r Supplementary Table 1}

#To Do
# NYHA CLASS is categorical - currently treated as continuous variable 
# Like the header in above : Number in each group in column names is hard coded. Need to be fixed 

SuppTable1 <- ACM %>%
  group_by(VarGroup) %>%
  summarise(
    #total=n(),
    #continuous Variables
	FINAL_EJECTION_FRACTION=continuousVariable_summary(FINAL_EJECTION_FRACTION),
	DIASTOLIC_DIAMETER_FINAL=continuousVariable_summary(DIASTOLIC_DIAMETER_FINAL),
	SYSTOLIC_DIAMETER=continuousVariable_summary(SYSTOLIC_DIAMETER),
	NYHA_CLASS=continuousVariable_summary(NYHA_CLASS),
    #binary Variables
	CURRENT_DRINKER=binaryVariable_summary(CURRENT_DRINKER,n()),
	DEATH=binaryVariable_summary(DEATH,n()),
	EJECTION_FRACTION_RECOVERY=binaryVariable_summary(EJECTION_FRACTION_RECOVERY,n()),	
	CRT=binaryVariable_summary(CRT,n()),
	QRS_gt_120=binaryVariable_summary(QRS_gt_120,n()),
	ICD=binaryVariable_summary(ICD,n()),
	TRANSPLANT=binaryVariable_summary(TRANSPLANT,n()),
	SMOKING=binaryVariable_summary(SMOKING,n()),
	HYPERTENSION=binaryVariable_summary(HYPERTENSION,n()),
	DYSLIPIDEMIA=binaryVariable_summary(DYSLIPIDEMIA,n()),
	RHYTHM=binaryVariable_summary(RHYTHM,n()),	
	SITUATION_INITIAL_EVALUATION=binaryVariable_summary(SITUATION_INITIAL_EVALUATION,n()),
	DIABETES_MELLITUS=binaryVariable_summary(DIABETES_MELLITUS,n()),
	COPD=binaryVariable_summary(COPD,n())
    ) 
	
SuppTable2 <- ACM %>%
  group_by(VarGroup) %>%
  summarise(
	FINAL_EJECTION_FRACTION=sum(is.na(FINAL_EJECTION_FRACTION)),
	DIASTOLIC_DIAMETER_FINAL=sum(is.na(DIASTOLIC_DIAMETER_FINAL)),
	SYSTOLIC_DIAMETER=sum(is.na(SYSTOLIC_DIAMETER)),
	NYHA_CLASS=sum(is.na(NYHA_CLASS)),
	CURRENT_DRINKER=sum(is.na(CURRENT_DRINKER)),
	DEATH=sum(is.na(DEATH)),
	EJECTION_FRACTION_RECOVERY=sum(is.na(EJECTION_FRACTION_RECOVERY)),	
	CRT=sum(is.na(CRT)),
	QRS_gt_120=sum(is.na(QRS_gt_120)),
	ICD=sum(is.na(ICD)),
	TRANSPLANT=sum(is.na(TRANSPLANT)),
	SMOKING=sum(is.na(SMOKING)),
	HYPERTENSION=sum(is.na(HYPERTENSION)),
	DYSLIPIDEMIA=sum(is.na(DYSLIPIDEMIA)),
	RHYTHM=sum(is.na(RHYTHM)),	
	SITUATION_INITIAL_EVALUATION=sum(is.na(SITUATION_INITIAL_EVALUATION)),
	DIABETES_MELLITUS=sum(is.na(DIABETES_MELLITUS)),
	COPD=sum(is.na(COPD))
    ) 
	
SuppTable3 <- ACM %>%
  filter(VarGroup=="TTNtv" | VarGroup=="GenotypeNegative") %>%
  summarise(
	FINAL_EJECTION_FRACTION=with(.,round(digits=4,wilcox.test(FINAL_EJECTION_FRACTION ~ VarGroup)$p.val)),
	DIASTOLIC_DIAMETER_FINAL=with(.,round(digits=4,wilcox.test(DIASTOLIC_DIAMETER_FINAL ~ VarGroup)$p.val)),
	SYSTOLIC_DIAMETER=with(.,round(digits=4,wilcox.test(SYSTOLIC_DIAMETER ~ VarGroup)$p.val)),
	NYHA_CLASS=with(.,round(digits=4,wilcox.test(NYHA_CLASS ~ VarGroup)$p.val)),
	
	CURRENT_DRINKER=with(., round(digits=4,fisher.test(x=CURRENT_DRINKER,y=VarGroup)$p.val)),
	DEATH=with(., round(digits=4,fisher.test(x=DEATH,y=VarGroup)$p.val)),
	EJECTION_FRACTION_RECOVERY=with(., round(digits=4,fisher.test(x=EJECTION_FRACTION_RECOVERY,y=VarGroup)$p.val)),	
	CRT=with(., round(digits=4,fisher.test(x=CRT,y=VarGroup)$p.val)),
	QRS_gt_120=with(., round(digits=4,fisher.test(x=QRS_gt_120,y=VarGroup)$p.val)),
	ICD=with(., round(digits=4,fisher.test(x=ICD,y=VarGroup)$p.val)),
	TRANSPLANT=with(., round(digits=4,fisher.test(x=TRANSPLANT,y=VarGroup)$p.val)),
	SMOKING=with(., round(digits=4,fisher.test(x=SMOKING,y=VarGroup)$p.val)),
	HYPERTENSION=with(., round(digits=4,fisher.test(x=HYPERTENSION,y=VarGroup)$p.val)),
	DYSLIPIDEMIA=with(., round(digits=4,fisher.test(x=DYSLIPIDEMIA,y=VarGroup)$p.val)),
	RHYTHM=with(., round(digits=4,fisher.test(x=RHYTHM,y=VarGroup)$p.val)),	
	SITUATION_INITIAL_EVALUATION=with(., round(digits=4,fisher.test(x=SITUATION_INITIAL_EVALUATION,y=VarGroup)$p.val)),
	DIABETES_MELLITUS=with(., round(digits=4,fisher.test(x=DIABETES_MELLITUS,y=VarGroup)$p.val)),
	COPD=with(., round(digits=4,fisher.test(x=COPD,y=VarGroup)$p.val))
    ) 
	

SuppTable1<-SuppTable1[,c("FINAL_EJECTION_FRACTION","DIASTOLIC_DIAMETER_FINAL","SYSTOLIC_DIAMETER","NYHA_CLASS","CURRENT_DRINKER","DEATH","EJECTION_FRACTION_RECOVERY","CRT","QRS_gt_120","ICD","TRANSPLANT","SMOKING","HYPERTENSION","DYSLIPIDEMIA","RHYTHM","SITUATION_INITIAL_EVALUATION","DIABETES_MELLITUS","COPD")]
SuppTable2<-SuppTable2[,c("FINAL_EJECTION_FRACTION","DIASTOLIC_DIAMETER_FINAL","SYSTOLIC_DIAMETER","NYHA_CLASS","CURRENT_DRINKER","DEATH","EJECTION_FRACTION_RECOVERY","CRT","QRS_gt_120","ICD","TRANSPLANT","SMOKING","HYPERTENSION","DYSLIPIDEMIA","RHYTHM","SITUATION_INITIAL_EVALUATION","DIABETES_MELLITUS","COPD")]
SuppTable3<-SuppTable3[,c("FINAL_EJECTION_FRACTION","DIASTOLIC_DIAMETER_FINAL","SYSTOLIC_DIAMETER","NYHA_CLASS","CURRENT_DRINKER","DEATH","EJECTION_FRACTION_RECOVERY","CRT","QRS_gt_120","ICD","TRANSPLANT","SMOKING","HYPERTENSION","DYSLIPIDEMIA","RHYTHM","SITUATION_INITIAL_EVALUATION","DIABETES_MELLITUS","COPD")]

SuppTable<-rbind(SuppTable1,SuppTable2,SuppTable3)
SuppTable<-t(SuppTable)

#colnames(SuppTable) <- c("DCMVariants_nonTTN[5]","GenotypeNegative[122]","TTNtv[14]","DCMVariants_nonTTN[NA's]","GenotypeNegative[NA's]","TTNtv[NA's]","P-value[TTNtv vs GenotypeNegative]")

SuppTable<-SuppTable[,c("3","2","1","6","5","4","7")]

colnames(SuppTable) <- c("TTNtv <br> (n=14)","Genotype Neg <br> (n=122)","Other Variants <br> (n=5)","TTNtv <br> (NA's)","Genotype Neg <br> (NA's)","Other Variants <br>(NA's)","P-value <br> (TTNtv vs Genotype Neg)")

#Make row names to lower case and remove underscore 
SuppTable_rowNames = rownames(SuppTable)
#replace all spaces and dots with underscore
SuppTable_rowNames<- gsub(x = SuppTable_rowNames,pattern = "_", replacement = " ",fixed = TRUE)
#SuppTable_rowNames<-tolower(SuppTable_rowNames)
rownames(SuppTable)<-SuppTable_rowNames

#pandoc.table(SuppTable, split.table = Inf)
# pander(SuppTable)
panderOptions('table.split.table', 800)
pander(SuppTable)
```


```{r stats, include=FALSE}

ACM %>%
  filter(VarGroup=="TTNtv" | VarGroup=="GenotypeNegative") %>%
  #with(.,wilcox.test(INITIAL_EJECTION_FRACTION ~ GENE))
  with(.,fisher.test(x=RHYTHM,y=VarGroup))

```

```{r Preparing for survival Analysis}

ACM_survival <- ACM[,c("BRUID","VarGroup","DATE_INITIAL_CLINICAL_ASSESSMENT","DATE_OF_LAST_CONSULTATION","DEATH","DATE_OF_DEATH","TRANSPLANT","DATE_OF_TRANSPLANT","Gr_alcohol","AGE_AT_INITIAL_CLINICAL_ASSESSMENT","INITIAL_EJECTION_FRACTION","DIASTOLIC_DIAMETER","SEX","ATRIAL_FIBRILLATION","FAMILY_HISTORY_OF_CARDIOMYOPATHY","FAMILY_HISTORY_OF_SUDDEN_CARDIAC_DEATH")]

#Add a column : Time (days)
ACM_survival<-cbind(ACM_survival,TIME="")
ACM_survival<-cbind(ACM_survival,EVENT="0")
ACM_survival<-cbind(ACM_survival,EndDate="")

ACM_survival$EndDate <- as.character(ACM_survival$EndDate)
ACM_survival$EVENT <- as.character(ACM_survival$EVENT)

#Death
ACM_survival$DATE_OF_DEATH <- as.character(ACM_survival$DATE_OF_DEATH)
endEvent <- ACM_survival$DEATH=="1"
ACM_survival[endEvent, "EndDate"] <- ACM_survival[endEvent, "DATE_OF_DEATH"]
ACM_survival[endEvent, "EVENT"] <- ACM_survival[endEvent, "DEATH"]

#Transplant
ACM_survival$DATE_OF_TRANSPLANT <- as.character(ACM_survival$DATE_OF_TRANSPLANT)
endEvent <- ACM_survival$TRANSPLANT=="1"
ACM_survival[endEvent, "EndDate"] <- ACM_survival[endEvent, "DATE_OF_TRANSPLANT"]
ACM_survival[endEvent, "EVENT"] <- ACM_survival[endEvent, "TRANSPLANT"]


#DATE_OF_LAST_CONSULTATION - for others
ACM_survival$DATE_OF_LAST_CONSULTATION <- as.character(ACM_survival$DATE_OF_LAST_CONSULTATION)
endEvent <- ACM_survival$EndDate==""
ACM_survival[endEvent, "EndDate"] <- ACM_survival[endEvent, "DATE_OF_LAST_CONSULTATION"]

#Calculate T: number of days between start and end point
ACM_survival$TIME <- as.Date(as.character(ACM_survival$EndDate), format="%Y-%m-%d")-
                  as.Date(as.character(ACM_survival$DATE_INITIAL_CLINICAL_ASSESSMENT), format="%Y-%m-%d")

ACM_survival <- ACM_survival[,c("BRUID","TIME","EVENT", "VarGroup","Gr_alcohol","AGE_AT_INITIAL_CLINICAL_ASSESSMENT","INITIAL_EJECTION_FRACTION","DIASTOLIC_DIAMETER","SEX","ATRIAL_FIBRILLATION","FAMILY_HISTORY_OF_CARDIOMYOPATHY","FAMILY_HISTORY_OF_SUDDEN_CARDIAC_DEATH")]

colnames(ACM_survival) <- c("BRUID","TIME","EVENT", "GENETIC_STATUS","Gr_alcohol","AGE_AT_INITIAL_CLINICAL_ASSESSMENT","INITIAL_EJECTION_FRACTION","DIASTOLIC_DIAMETER","SEX","ATRIAL_FIBRILLATION","FAMILY_HISTORY_OF_CARDIOMYOPATHY","FAMILY_HISTORY_OF_SUDDEN_CARDIAC_DEATH")

#write table
  write.table(ACM_survival, quote=FALSE, sep ="\t", row.names = F, file="../ACM_Survival_Analysis_input.txt") 

  #pander(ACM_survival)
```

##Table 2
```{r ACM vs DCM vs HVOL, results='asis'}

# To Do
# ? how to add ethnicity to table


DCM_demo <- read.table("../DCM_pheno.txt", header=T, sep='\t')
HVOL_demo <- read.table("../HVOL_pheno.txt", header=T, sep='\t')

#Add missing  columns to the tables 
DCM_demo <-cbind(DCM_demo, EF_echo=NA, DD_echo=NA)
HVOL_demo <-cbind(HVOL_demo, EF_echo=NA, DD_echo=NA)
ACM_demo <-cbind(group = 'ACM', ACM, Ethnicity ='C', EF_MRI=NA, DD_MRI=NA)


# recreate tables in correct order or columns 
DCM_demo <- DCM_demo[,c("group", "SEX","AGE_AT_scan","EF_echo","DD_echo","EF","DD","Ethnicity")]
HVOL_demo <- HVOL_demo[,c("group", "SEX","AGE_AT_scan","EF_echo","DD_echo","EF","DD","Ethnicity")]
ACM_demo <- ACM_demo[,c("group","SEX","AGE_AT_INITIAL_CLINICAL_ASSESSMENT","INITIAL_EJECTION_FRACTION","DIASTOLIC_DIAMETER","EF_MRI","DD_MRI","Ethnicity")]

#give uniform column names 
colnames(DCM_demo) <- c("group", "SEX","AGE_AT_scan","EF_echo","DD_echo","EF_MRI","DD_MRI","Ethnicity")
colnames(HVOL_demo)<- c("group", "SEX","AGE_AT_scan","EF_echo","DD_echo","EF_MRI","DD_MRI","Ethnicity")
colnames(ACM_demo) <- c("group", "SEX","AGE_AT_scan","EF_echo","DD_echo","EF_MRI","DD_MRI","Ethnicity")

#dim(DCM_demo)
#dim(HVOL_demo)
#dim(ACM_demo)

ALLgroups <-rbind(ACM_demo,DCM_demo,HVOL_demo)

#make sex 0/1 for easier analysis 
ALLgroups$SEX[ALLgroups$SEX == "M"] <- 1 
ALLgroups$SEX[ALLgroups$SEX == "F"] <- 0 
ALLgroups$SEX[ALLgroups$SEX == 2] <- 0 
ALLgroups$SEX <- as.numeric(ALLgroups$SEX)

#make ethnicity 0/1 for easier analysis 
ALLgroups$Ethnicity <- as.character(ALLgroups$Ethnicity)
ALLgroups$Ethnicity[ALLgroups$Ethnicity != "C"] <- 0 
ALLgroups$Ethnicity[ALLgroups$Ethnicity == "C"] <- 1 
ALLgroups$Ethnicity <- as.numeric(ALLgroups$Ethnicity)

GroupTable1 <- ALLgroups %>%
group_by(group) %>%
  summarise(
    #total=n(),
    #continuous Variables
    AGE_AT_scan=continuousVariable_summary(AGE_AT_scan),
    EF_echo=continuousVariable_summary(EF_echo),
    DD_echo=continuousVariable_summary(DD_echo),
    EF_MRI=continuousVariable_summary(EF_MRI),
    DD_MRI=continuousVariable_summary(DD_MRI),    
    #binary Variables
    SEX=binaryVariable_summary(SEX,n()),
    Ethnicity=binaryVariable_summary(Ethnicity,n())
    
    ) 

GroupTable2 <- ALLgroups %>%
  group_by(group) %>%
  summarise(
	AGE_AT_scan=sum(is.na(AGE_AT_scan)),
	EF_echo=sum(is.na(EF_echo)),
	DD_echo=sum(is.na(DD_echo)),
	EF_MRI=sum(is.na(EF_MRI)),
	DD_MRI=sum(is.na(DD_MRI)),	
	SEX=sum(is.na(SEX)),
	Ethnicity=sum(is.na(Ethnicity))
    ) 

GroupTable1<-GroupTable1[,c("AGE_AT_scan","EF_echo","DD_echo","EF_MRI","DD_MRI","SEX","Ethnicity")]
GroupTable2<-GroupTable2[,c("AGE_AT_scan","EF_echo","DD_echo","EF_MRI","DD_MRI","SEX","Ethnicity")]

GroupTable<-rbind(GroupTable1,GroupTable2)

#remove statistics based on empty columns 
GroupTable$EF_echo[GroupTable$EF_echo == "NaN ± NA" | GroupTable$EF_echo == "141" | GroupTable$EF_echo == "366" | GroupTable$EF_echo == "445"] <- "-"
GroupTable$DD_echo[GroupTable$DD_echo == "NaN ± NA" | GroupTable$DD_echo == "141" | GroupTable$DD_echo == "366" | GroupTable$DD_echo == "445"] <- "-"
GroupTable$EF_MRI[GroupTable$EF_MRI == "NaN ± NA" | GroupTable$EF_MRI == "141" | GroupTable$EF_MRI == "366" | GroupTable$EF_MRI == "445"] <- "-"
GroupTable$DD_MRI[GroupTable$DD_MRI == "NaN ± NA" | GroupTable$DD_MRI == "141" | GroupTable$DD_MRI == "366" | GroupTable$DD_MRI == "445"] <- "-"

GroupTable<-t(GroupTable)

colnames(GroupTable) <- c("ACM <br> (n=141)","DCM <br> (n=366)","HVOL <br> (n=445)","ACM <br> (NA's)","DCM <br> (NA's)","HVOL <br> (NA's)")
rownames(GroupTable) <- c("Age at scan", "Left ventricular ejection fraction (echo)","Left ventricular end diastolic diameter (echo)", "Left ventricular ejection fraction (MRI)","Left ventricular end diastolic volume (MRI)","Sex (male)","Ethnicity (caucasian)")




panderOptions('table.split.table', 800)
pander(GroupTable)


```



## Table 3 
```{r ACM - recovery stable severe}

ACM_recov_temp <- ACM[,c("BRUID","VarGroup","DEATH","TRANSPLANT","EJECTION_FRACTION_RECOVERY")]

ACM_recov <-cbind(ACM_recov_temp,Death_or_Transplant=0,LVEF_recovery=0,No_LVEF_Recovery=0)

ACM_recov$Death_or_Transplant[ACM_recov$DEATH=="1"] <- 1
ACM_recov$Death_or_Transplant[ACM_recov$TRANSPLANT=="1"] <- 1

ACM_recov$LVEF_recovery[ACM_recov$EJECTION_FRACTION_RECOVERY==1] <- 1
ACM_recov$LVEF_recovery[is.na(ACM_recov$EJECTION_FRACTION_RECOVERY)] <- NA
ACM_recov$LVEF_recovery[ACM_recov$Death_or_Transplant==1] <- 0

ACM_recov$No_LVEF_Recovery[ACM_recov$EJECTION_FRACTION_RECOVERY==0] <- 1
ACM_recov$No_LVEF_Recovery[is.na(ACM_recov$EJECTION_FRACTION_RECOVERY)] <- NA
ACM_recov$No_LVEF_Recovery[ACM_recov$Death_or_Transplant==1] <- 0


RecoveryTable <- ACM_recov %>%
group_by(VarGroup) %>%
  summarise(
    #binary Variables
    Death_or_Transplant=binaryVariable_summary(Death_or_Transplant,n()),
    LVEF_recovery=binaryVariable_summary(LVEF_recovery,n()),    
    No_LVEF_Recovery=binaryVariable_summary(No_LVEF_Recovery,n())
    ) 


RecoveryTable<-RecoveryTable[,c("Death_or_Transplant","LVEF_recovery","No_LVEF_Recovery")]

RecoveryTable<- t(RecoveryTable)

colnames(RecoveryTable) <- c("Other Variants <br> (n=5)","Genotype Neg <br> (n=122)","TTNtv <br> (n=14)")
RecoveryTable<-RecoveryTable[,c("TTNtv <br> (n=14)","Genotype Neg <br> (n=122)","Other Variants <br> (n=5)")]


#Make row names to lower case and remove underscore 
RecoveryTable_rowNames = rownames(RecoveryTable)
#replace all spaces and dots with underscore
RecoveryTable_rowNames<- gsub(x = RecoveryTable_rowNames,pattern = "_", replacement = " ",fixed = TRUE)
rownames(RecoveryTable)<-RecoveryTable_rowNames



panderOptions('table.split.table', 800)
pander(RecoveryTable)

```

