---
title: "All Tables"
author: "Risha Govind"
date: "`r Sys.time()`"
output:
  # html_document
  word_document:
    reference_docx: ./templateLandscape.docx
---
  
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning=FALSE, size="smallsize")
library(lazyeval)
library(plyr)
library(dplyr)
library(pander)
library(gdata)
library(gtools)

 

```


```{r load data}
## Load datasets

#ACM pheno 
ACM <-read.table("../input-files/ACM_pheno.tsv", header=T, sep='\t')

ACM_survival_raw <- read.table("../input-files/ACM_Survival_Analysis_raw.txt", header=T, sep='\t') 


#Names Of columns and Rows as to appear in manuscript
Variable_Names_for_manuscript = read.table("../input-files/Variable_Names_for_manuscript.txt", header=FALSE, sep='\t')

#Demographics/phenotype data on DCM and HVOL
DCM_demo <- read.table("../input-files/DCM_pheno.txt", header=T, sep='\t')
HVOL_demo <- read.table("../input-files/HVOL_pheno.txt", header=T, sep='\t')

#Rare Variants Data
RareVarRaw <- read.table("../input-files/Filtered_Rare_Variants.txt", header=T) 

#List of Genes & VarTypes to Burden Test on
Gene_VarType_BurdenTest_list <- read.table("../input-files/Gene_VarType_BurdenTest.txt", header=F)

#Sample IDs & cohorts
SampleIDs <- read.table("../input-files/Filtered_SampleIDs.txt", header=T) 

#ExAv frequencies from Roddy
GeneVartype_ExAC_freqs <- read.table("../input-files/GeneVartype_ExAC_freqs.txt", header=T,sep='\t') 


#dim(ACM_pheno)
```


```{r functions}

# Rename column names directly. Don't concert to array first. :  names(d)[names(d)=="beta"] <- "two"
# Uppercase first alphabet only : firstup("i like pizza very much")

firstup <- function(x) {
   substr(x, 1, 1) <- toupper(substr(x, 1, 1))
x
}

update_Variable_Names_for_manuscript <- function(x){
  
  x<-as.data.frame(x)

  count <- nrow(Variable_Names_for_manuscript)
  
  for (i in 1:count){
    headerOld <- Variable_Names_for_manuscript$V1[i]
    headerNew <- Variable_Names_for_manuscript$V2[i]
    
    #colnames(x)[colnames(x)==headerOld] <- headerNew
    #rownames(x)[rownames(x)==headerOld] <- headerNew
    colnames(x) <- gsub(headerOld, headerNew, colnames(x))
    rownames(x) <- gsub(headerOld, headerNew, rownames(x))    
  }
  
  colnames(x) <- gsub("\\_", " ", colnames(x))
  rownames(x) <- gsub("\\_", " ", rownames(x))

  return(x)
}


continuousVariable_summary <- function(x){
  c<-paste(round(mean(x,na.rm=T), digits=1)," \u00b1 ", round(sd(x,na.rm=T), digits=1), sep="")
  return(c)
}

binaryVariable_summary <- function(e,f){
  b<-paste(round(sum(e,na.rm=T))," \u0028",round(((sum(e,na.rm=T)/f) * 100), digits=1),"\u0025\u0029", sep="")
  return(b)
}


Summary_burden_Analysis <- function(e,f){
  a <- e %>%
    summarise(
    ALL_GENES=with(., signif(digits=2,fisher.test(x=ALL_GENES,y=group)$p.val)),
    TTN_TV=with(., signif(digits=2,fisher.test(x=TTN_TV,y=group)$p.val)),
    ALL_GENES_except_TTN=with(., signif(digits=2,fisher.test(x=ALL_GENES_except_TTN,y=group)$p.val))
    )
  rownames(a)<-f
  return(a)
}

All_burden_Analysis <- function(e,f){

  #remove columns that have all zeros in the groups we are comparing - fisher.test gives error 
  e<-e[, colSums(e != 0) > 0]

  a <- e %>%
    summarise_each(funs(signif(digits=2,fisher.test(x=.,y=group)$p.val)), vars = BAG3_non_TV:ALL_GENES)  #BAG3_non_TV & ALL_GENES are hardcoded !!! 

  #update colnames
  colnames(a)<-colnames(e[-1])
  rownames(a)<-f

  #a<-t(a)
  return(a)
}

RBIND <- function(datalist) {
  temp <- rbind.fill(datalist)
  rownames(temp) <- unlist(lapply(datalist, row.names))
  temp
}


```



```{r Get Mean and Std.Dev}

#To Do
# AGE oF SYMPTOM
# NYHA CLASS 
# Did `na.rm=T` here, because there was some missing data. Need to check if this appropriate.

MainTable1 <- ACM %>%
  group_by(VarGroup) %>%
  summarise(
    #total=n(),
    #continuous Variables
    Alcohol_grams=continuousVariable_summary(Alcohol_grams),
    Age_at_initial_clinical_assesement=continuousVariable_summary(Age_at_initial_clinical_assesement),
    Initial_ejection_fraction=continuousVariable_summary(Initial_ejection_fraction),
    Diastolic_diameter=continuousVariable_summary(Diastolic_diameter),
    #binary Variables
    Sex_males=binaryVariable_summary(Sex_males,n()),
    Atrial_fibrillation=binaryVariable_summary(Atrial_fibrillation,n()),
    Family_history_of_CM=binaryVariable_summary(Family_history_of_CM,n()),
    Family_history_of_SCD=binaryVariable_summary(Family_history_of_SCD,n())
  ) 

```



```{r get number of NAs}
MainTable2 <- ACM %>%
  group_by(VarGroup) %>%
  summarise(
    Alcohol_grams=sum(is.na(Alcohol_grams)),
    Age_at_initial_clinical_assesement=sum(is.na(Age_at_initial_clinical_assesement)),
    Initial_ejection_fraction=sum(is.na(Initial_ejection_fraction)),
    Diastolic_diameter=sum(is.na(Diastolic_diameter)),
    Sex_males=sum(is.na(Sex_males)),
    Atrial_fibrillation=sum(is.na(Atrial_fibrillation)),
    Family_history_of_CM=sum(is.na(Family_history_of_CM)),
    Family_history_of_SCD=sum(is.na(Family_history_of_SCD))
  ) 

```

```{r get P-values}
#Mann-Whitney or wilcox.test : continuous (not normal) vs categorical (binary) 
#Fisher Test : categorical (binary) vs categorical (binary) 


MainTable3 <- ACM %>%
  filter(VarGroup=="TTNtv" | VarGroup=="GenotypeNegative") %>%
  summarise(
    Alcohol_grams=with(.,signif(digits=2,wilcox.test(Alcohol_grams ~ VarGroup)$p.val)),
    Age_at_initial_clinical_assesement=with(.,signif(digits=2,wilcox.test(Age_at_initial_clinical_assesement ~ VarGroup)$p.val)),
    Initial_ejection_fraction=with(.,signif(digits=2,wilcox.test(Initial_ejection_fraction ~ VarGroup)$p.val)),
    Diastolic_diameter=with(.,signif(digits=2,wilcox.test(Diastolic_diameter ~ VarGroup)$p.val)),
    Sex_males=with(., signif(digits=2,fisher.test(x=Sex_males,y=VarGroup)$p.val)),
    Atrial_fibrillation=with(., signif(digits=2,fisher.test(x=Atrial_fibrillation,y=VarGroup)$p.val)),
    Family_history_of_CM=with(., signif(digits=2,fisher.test(x=Family_history_of_CM,y=VarGroup)$p.val)),
    Family_history_of_SCD=with(., signif(digits=2,fisher.test(x=Family_history_of_SCD,y=VarGroup)$p.val))
  )

#Add VarGroup column to MainTable3
MainTable3<-cbind(VarGroup="pvalue_TTN_GN",MainTable3)

```


```{r Combine all above : Characteristics of ACM cases with and without titin truncating variants table}

MainTable<-rbind(MainTable1,MainTable3)

MainTable<-as.data.frame(t(MainTable))

#Make first row column names
names(MainTable) <- lapply(MainTable[1, ], as.character)
MainTable <- MainTable[-1,] 

#Update Column names and row names
MainTable<-update_Variable_Names_for_manuscript(MainTable)

#change order of columns
MainTable<-MainTable[,c(3,2,1,4)]


#Table 3 : Characteristics of ACM cases with and without titin truncating variants  
Characteristics_ACM_wrt_TTNTv<-as.data.frame(MainTable)


```

```{r Table for internal use - missing data info for above table}

MainTable_NA<-as.data.frame(t(MainTable2))

#Make first row column names
names(MainTable_NA) <- lapply(MainTable_NA[1, ], as.character)
MainTable_NA <- MainTable_NA[-1,] 

#Update Column names and row names
MainTable_NA<-update_Variable_Names_for_manuscript(MainTable_NA)

#change order of columns
MainTable_NA<-MainTable_NA[,c(3,2,1)]

#Table 3 : Characteristics of ACM cases with and without titin truncating variants  
Characteristics_ACM_wrt_TTNTv_NAs<-as.data.frame(MainTable_NA)


```

```{r Supplementary Table : Clinical characteristics of ACM cohort based on genotype}

 
    # NYHA_CLASS=paste(paste(sep="",length(NYHA_CLASS[NYHA_CLASS==1])," \u0028",round((length(NYHA_CLASS[NYHA_CLASS==1])/n() * 100), digits=1),"\u0025\u0029"),
    #                  paste(sep="",length(NYHA_CLASS[NYHA_CLASS==2])," \u0028",round((length(NYHA_CLASS[NYHA_CLASS==2])/n() * 100), digits=1),"\u0025\u0029"),
    #                  paste(sep="",length(NYHA_CLASS[NYHA_CLASS==3])," \u0028",round((length(NYHA_CLASS[NYHA_CLASS==3])/n() * 100), digits=1),"\u0025\u0029"),
    #                  paste(sep="",length(NYHA_CLASS[NYHA_CLASS==4])," \u0028",round((length(NYHA_CLASS[NYHA_CLASS==4])/n() * 100), digits=1),"\u0025\u0029"),
    #                  sep="  /  "),


SuppTable1 <- ACM %>%
  group_by(VarGroup) %>%
  summarise(
    #total=n(),
    #continuous Variables
    Initial_ejection_fraction=continuousVariable_summary(Initial_ejection_fraction),
    FINAL_EJECTION_FRACTION=continuousVariable_summary(FINAL_EJECTION_FRACTION),
    Diastolic_diameter=continuousVariable_summary(Diastolic_diameter),
    Diastolic_diameter_FINAL=continuousVariable_summary(Diastolic_diameter_FINAL),
    NYHA_CLASS=paste(sep="/",length(NYHA_CLASS[NYHA_CLASS==1]),length(NYHA_CLASS[NYHA_CLASS==2]),length(NYHA_CLASS[NYHA_CLASS==3]),length(NYHA_CLASS[NYHA_CLASS==4])),
    #binary Variables 
    CURRENT_DRINKER=binaryVariable_summary(CURRENT_DRINKER,n()),
    NON_DRINKER=binaryVariable_summary(NON_DRINKER,n()),
    DEATH=binaryVariable_summary(DEATH,n()),
    EJECTION_FRACTION_RECOVERY=binaryVariable_summary(EJECTION_FRACTION_RECOVERY,(n()-sum(is.na(EJECTION_FRACTION_RECOVERY)))),	
    CRT=binaryVariable_summary(CRT,n()),
    QRS_gt_120=binaryVariable_summary(QRS_gt_120,(n()-sum(is.na(QRS_gt_120)))),
    ICD=binaryVariable_summary(ICD,n()),
    TRANSPLANT=binaryVariable_summary(TRANSPLANT,n()),
    SMOKING=binaryVariable_summary(SMOKING,n()),
    HYPERTENSION=binaryVariable_summary(HYPERTENSION,n()),
    DYSLIPIDEMIA=binaryVariable_summary(DYSLIPIDEMIA,n()),
    RHYTHM=binaryVariable_summary(RHYTHM,n()),	
    SITUATION_INITIAL_EVALUATION=binaryVariable_summary(SITUATION_INITIAL_EVALUATION,n()),
    DIABETES_MELLITUS=binaryVariable_summary(DIABETES_MELLITUS,n()),
    COPD=binaryVariable_summary(COPD,n())
  ) 


SuppTable2 <- ACM %>%
  group_by(VarGroup) %>%
  summarise(
    Initial_ejection_fraction=sum(is.na(Initial_ejection_fraction)),
    FINAL_EJECTION_FRACTION=sum(is.na(FINAL_EJECTION_FRACTION)),
    Diastolic_diameter=sum(is.na(Diastolic_diameter)),
    Diastolic_diameter_FINAL=sum(is.na(Diastolic_diameter_FINAL)),
    NYHA_CLASS=sum(is.na(NYHA_CLASS)),
    CURRENT_DRINKER=sum(is.na(CURRENT_DRINKER)),
    NON_DRINKER=sum(is.na(NON_DRINKER)),
    DEATH=sum(is.na(DEATH)),
    EJECTION_FRACTION_RECOVERY=sum(is.na(EJECTION_FRACTION_RECOVERY)),	
    CRT=sum(is.na(CRT)),
    QRS_gt_120=sum(is.na(QRS_gt_120)),
    ICD=sum(is.na(ICD)),
    TRANSPLANT=sum(is.na(TRANSPLANT)),
    SMOKING=sum(is.na(SMOKING)),
    HYPERTENSION=sum(is.na(HYPERTENSION)),
    DYSLIPIDEMIA=sum(is.na(DYSLIPIDEMIA)),
    RHYTHM=sum(is.na(RHYTHM)),	
    SITUATION_INITIAL_EVALUATION=sum(is.na(SITUATION_INITIAL_EVALUATION)),
    DIABETES_MELLITUS=sum(is.na(DIABETES_MELLITUS)),
    COPD=sum(is.na(COPD))
  ) 

SuppTable3 <- ACM %>%
  filter(VarGroup=="TTNtv" | VarGroup=="GenotypeNegative") %>%
  summarise(
    Initial_ejection_fraction=with(.,signif(digits=2,wilcox.test(Initial_ejection_fraction ~ VarGroup)$p.val)),
    FINAL_EJECTION_FRACTION=with(.,signif(digits=2,wilcox.test(FINAL_EJECTION_FRACTION ~ VarGroup)$p.val)),
    Diastolic_diameter=with(.,signif(digits=2,wilcox.test(Diastolic_diameter ~ VarGroup)$p.val)),
    Diastolic_diameter_FINAL=with(.,signif(digits=2,wilcox.test(Diastolic_diameter_FINAL ~ VarGroup)$p.val)),
    NYHA_CLASS=with(.,signif(digits=2,wilcox.test(NYHA_CLASS ~ VarGroup)$p.val)),
    CURRENT_DRINKER=with(., signif(digits=2,fisher.test(x=CURRENT_DRINKER,y=VarGroup)$p.val)),
    NON_DRINKER=with(., signif(digits=2,fisher.test(x=NON_DRINKER,y=VarGroup)$p.val)),
    DEATH=with(., signif(digits=2,fisher.test(x=DEATH,y=VarGroup)$p.val)),
    EJECTION_FRACTION_RECOVERY=with(., signif(digits=2,fisher.test(x=EJECTION_FRACTION_RECOVERY,y=VarGroup)$p.val)),	
    CRT=with(., signif(digits=2,fisher.test(x=CRT,y=VarGroup)$p.val)),
    QRS_gt_120=with(., signif(digits=2,fisher.test(x=QRS_gt_120,y=VarGroup)$p.val)),
    ICD=with(., signif(digits=2,fisher.test(x=ICD,y=VarGroup)$p.val)),
    TRANSPLANT=with(., signif(digits=2,fisher.test(x=TRANSPLANT,y=VarGroup)$p.val)),
    SMOKING=with(., signif(digits=2,fisher.test(x=SMOKING,y=VarGroup)$p.val)),
    HYPERTENSION=with(., signif(digits=2,fisher.test(x=HYPERTENSION,y=VarGroup)$p.val)),
    DYSLIPIDEMIA=with(., signif(digits=2,fisher.test(x=DYSLIPIDEMIA,y=VarGroup)$p.val)),
    RHYTHM=with(., signif(digits=2,fisher.test(x=RHYTHM,y=VarGroup)$p.val)),	
    SITUATION_INITIAL_EVALUATION=with(., signif(digits=2,fisher.test(x=SITUATION_INITIAL_EVALUATION,y=VarGroup)$p.val)),
    DIABETES_MELLITUS=with(., signif(digits=2,fisher.test(x=DIABETES_MELLITUS,y=VarGroup)$p.val)),
    COPD=with(., signif(digits=2,fisher.test(x=COPD,y=VarGroup)$p.val))
  ) 



SuppTable1<-SuppTable1[,c("NYHA_CLASS","Diastolic_diameter","Diastolic_diameter_FINAL","Initial_ejection_fraction","FINAL_EJECTION_FRACTION","EJECTION_FRACTION_RECOVERY","DEATH","TRANSPLANT","CURRENT_DRINKER","NON_DRINKER","CRT","QRS_gt_120","ICD","SMOKING","HYPERTENSION","DYSLIPIDEMIA","DIABETES_MELLITUS","COPD")]
SuppTable2<-SuppTable2[,c("NYHA_CLASS","Diastolic_diameter","Diastolic_diameter_FINAL","Initial_ejection_fraction","FINAL_EJECTION_FRACTION","EJECTION_FRACTION_RECOVERY","DEATH","TRANSPLANT","CURRENT_DRINKER","NON_DRINKER","CRT","QRS_gt_120","ICD","SMOKING","HYPERTENSION","DYSLIPIDEMIA","DIABETES_MELLITUS","COPD")]
SuppTable3<-SuppTable3[,c("NYHA_CLASS","Diastolic_diameter","Diastolic_diameter_FINAL","Initial_ejection_fraction","FINAL_EJECTION_FRACTION","EJECTION_FRACTION_RECOVERY","DEATH","TRANSPLANT","CURRENT_DRINKER","NON_DRINKER","CRT","QRS_gt_120","ICD","SMOKING","HYPERTENSION","DYSLIPIDEMIA","DIABETES_MELLITUS","COPD")]

SuppTable<-rbind(SuppTable1,SuppTable2,SuppTable3)
SuppTable<-t(SuppTable)

SuppTable<-SuppTable[,c("3","2","1","6","5","4","7")]

SuppTable<-update_Variable_Names_for_manuscript(SuppTable)

colnames(SuppTable) <- c("TTNtv (n=14)","Genotype Neg (n=122)","Other Variants (n=5)","TTNtv (NA's)","Genotype Neg (NA's)","Other Variants (NA's)","P-value (TTNtv vs Genotype Neg)")

##Make row names to lower case and remove underscore 
#SuppTable_rowNames = rownames(SuppTable)
##replace all spaces and dots with underscore
#SuppTable_rowNames<- gsub(x = SuppTable_rowNames,pattern = "_", replacement = " ",fixed = TRUE)
##SuppTable_rowNames<-tolower(SuppTable_rowNames)
#rownames(SuppTable)<-SuppTable_rowNames

#Supplemetary Table : Clinical characteristics of ACM cohort based on genotype

SuppTable_clinical_Characteristics_ACM<-SuppTable

```


```{r stats, include=FALSE}

ACM %>%
  filter(VarGroup=="TTNtv" | VarGroup=="GenotypeNegative") %>%
  #with(.,wilcox.test(Initial_ejection_fraction ~ GENE))
  with(.,fisher.test(x=RHYTHM,y=VarGroup))

```

```{r ACM vs DCM vs HVOL, results='asis'}


#Add missing  columns to the tables 
DCM_demo <-cbind(DCM_demo, EF_echo=NA, DD_echo=NA)
HVOL_demo <-cbind(HVOL_demo, EF_echo=NA, DD_echo=NA)
ACM_demo <-cbind(group = 'ACM', ACM, Ethnicity ='C', EF_MRI=NA, DD_MRI=NA)

# recreate tables in correct order or columns 
DCM_demo <- DCM_demo[,c("group", "SEX","AGE_AT_scan","EF_echo","DD_echo","EF","DD","Ethnicity")]
HVOL_demo <- HVOL_demo[,c("group", "SEX","AGE_AT_scan","EF_echo","DD_echo","EF","DD","Ethnicity")]
ACM_demo <- ACM_demo[,c("group","Sex_males","Age_at_initial_clinical_assesement","Initial_ejection_fraction","Diastolic_diameter","EF_MRI","DD_MRI","Ethnicity")]

#give uniform column names 
colnames(DCM_demo) <- c("group", "Sex_males","AGE_AT_scan","EF_echo","DD_echo","EF_MRI","DD_MRI","Ethnicity")
colnames(HVOL_demo)<- c("group", "Sex_males","AGE_AT_scan","EF_echo","DD_echo","EF_MRI","DD_MRI","Ethnicity")
colnames(ACM_demo) <- c("group", "Sex_males","AGE_AT_scan","EF_echo","DD_echo","EF_MRI","DD_MRI","Ethnicity")

#dim(DCM_demo)
#dim(HVOL_demo)
#dim(ACM_demo)

ALLgroups <-rbind(ACM_demo,DCM_demo,HVOL_demo)

#make Sex_males 0/1 for easier analysis 
ALLgroups$Sex_males[ALLgroups$Sex_males == "M"] <- 1 
ALLgroups$Sex_males[ALLgroups$Sex_males == "F"] <- 0 
ALLgroups$Sex_males[ALLgroups$Sex_males == 2] <- 0 
ALLgroups$Sex_males <- as.numeric(ALLgroups$Sex_males)

#make ethnicity 0/1 for easier analysis 
ALLgroups$Ethnicity <- as.character(ALLgroups$Ethnicity)
ALLgroups$Ethnicity[ALLgroups$Ethnicity != "C"] <- 0 
ALLgroups$Ethnicity[ALLgroups$Ethnicity == "C"] <- 1 
ALLgroups$Ethnicity <- as.numeric(ALLgroups$Ethnicity)

GroupTable1 <- ALLgroups %>%
  group_by(group) %>%
  summarise(
    #total=n(),
    #continuous Variables
    AGE_AT_scan=continuousVariable_summary(AGE_AT_scan),
    EF_echo=continuousVariable_summary(EF_echo),
    EF_MRI=continuousVariable_summary(EF_MRI),
    DD_echo=continuousVariable_summary(DD_echo),
    DD_MRI=continuousVariable_summary(DD_MRI),    
    #binary Variables
    Sex_males=binaryVariable_summary(Sex_males,n()),
    Ethnicity=binaryVariable_summary(Ethnicity,n())
    
  ) 

GroupTable2 <- ALLgroups %>%
  filter(group=="ACM" | group=="DCM") %>%
  summarise(
    #continuous Variables
    AGE_AT_scan=with(.,signif(digits=2,wilcox.test(AGE_AT_scan ~ group)$p.val)),
    EF_echo="-",
    EF_MRI="-",
    DD_echo="-",
    DD_MRI="-",
    #binary Variables
    Sex_males=with(., signif(digits=2,fisher.test(x=Sex_males,y=group)$p.val)),  
    Ethnicity=1    
  )


GroupTable3 <- ALLgroups %>%
  filter(group=="ACM" | group=="HVOL") %>%
  summarise(
    #continuous Variables
    AGE_AT_scan=with(.,signif(digits=2,wilcox.test(AGE_AT_scan ~ group)$p.val)),
    EF_echo="-",
    EF_MRI="-",
    DD_echo="-",	
    DD_MRI="-",	
    #binary Variables
    Sex_males=with(., signif(digits=2,fisher.test(x=Sex_males,y=group)$p.val)), 
    Ethnicity=1   
  )


GroupTable4 <- ALLgroups %>%
  filter(group=="DCM" | group=="HVOL") %>%
  summarise(
    #continuous Variables
    AGE_AT_scan=with(.,signif(digits=2,wilcox.test(AGE_AT_scan ~ group)$p.val)),
    EF_echo="-",
    EF_MRI=with(.,signif(digits=2,wilcox.test(EF_MRI ~ group)$p.val)),
    DD_echo="-",
    DD_MRI=with(.,signif(digits=2,wilcox.test(DD_MRI ~ group)$p.val)),	
    #binary Variables
    Sex_males=with(., signif(digits=2,fisher.test(x=Sex_males,y=group)$p.val)), 
    Ethnicity=1 
  )

GroupTable2<-cbind(group="pvalue_A_H",GroupTable2)
GroupTable3<-cbind(group="pvalue_D_H",GroupTable3)
GroupTable4<-cbind(group="pvalue_A_D",GroupTable4)

```

#Table 1 : Clinical characteristics of patient cohorts 

```{r Combine all above : Clinical characteristics of patient cohorts}

GroupTable<-rbind(GroupTable1)

#remove statistics based on empty columns 
GroupTable$EF_echo[GroupTable$EF_echo == "NaN ± NA"] <- "-"
GroupTable$DD_echo[GroupTable$DD_echo == "NaN ± NA"] <- "-"
GroupTable$EF_MRI[GroupTable$EF_MRI == "NaN ± NA"] <- "-"
GroupTable$DD_MRI[GroupTable$DD_MRI == "NaN ± NA"] <- "-"


GroupTable<-as.data.frame(t(GroupTable))

#Make first row column names
names(GroupTable) <- lapply(GroupTable[1, ], as.character)
GroupTable <- GroupTable[-1,] 

#Update Column names and row names
#GroupTable<-update_colnames_VarGroups(GroupTable)
#GroupTable<-update_rownames_characteristics(GroupTable)
GroupTable<-update_Variable_Names_for_manuscript(GroupTable)

#change order of columns
GroupTable<-GroupTable[,c(1,2,3)]

panderOptions('table.split.table', 800)
pander(GroupTable)


```


#### Table for internal use - additional info for above table (missing data info and stats test)
```{r missing data info for above table}

GroupTable <- ALLgroups %>%
  group_by(group) %>%
  summarise(
    AGE_AT_scan=sum(is.na(AGE_AT_scan)),
    EF_echo=sum(is.na(EF_echo)),
    EF_MRI=sum(is.na(EF_MRI)),
    DD_echo=sum(is.na(DD_echo)),
    DD_MRI=sum(is.na(DD_MRI)),	
    Sex_males=sum(is.na(Sex_males)),
    Ethnicity=sum(is.na(Ethnicity))
  ) 

#remove statistics based on empty columns 
GroupTable$EF_echo[GroupTable$EF_echo == "141" | GroupTable$EF_echo == "366" | GroupTable$EF_echo == "445"] <- "-"
GroupTable$DD_echo[GroupTable$DD_echo == "141" | GroupTable$DD_echo == "366" | GroupTable$DD_echo == "445"] <- "-"
GroupTable$EF_MRI[GroupTable$EF_MRI == "141" | GroupTable$EF_MRI == "366" | GroupTable$EF_MRI == "445"] <- "-"
GroupTable$DD_MRI[GroupTable$DD_MRI == "141" | GroupTable$DD_MRI == "366" | GroupTable$DD_MRI == "445"] <- "-"

GroupTable<-rbind(GroupTable,GroupTable2,GroupTable3,GroupTable4)

GroupTable<-t(GroupTable)

#Make first row column names
colnames(GroupTable) <- GroupTable[1, ]
GroupTable <- GroupTable[-1,] 

GroupTable<-update_Variable_Names_for_manuscript(GroupTable)

colnames(GroupTable) <- gsub(".*141*)", "NA's:ACM", colnames(GroupTable))
colnames(GroupTable) <- gsub(".*366*)", "NA's:DCM", colnames(GroupTable))
colnames(GroupTable) <- gsub(".*445*)", "NA's:HVOL", colnames(GroupTable))

panderOptions('table.split.table', 800)
pander(GroupTable)

```

```{r ACM - recovery stable severe}

ACM_recov_temp <- ACM_survival_raw[,c("BRUID","VarGroup","DEATH","TRANSPLANT","EJECTION_FRACTION_RECOVERY","TIME")]

ACM_recov <-cbind(ACM_recov_temp,Death_or_Transplant=0,LVEF_recovery=0,No_LVEF_Recovery=0)

ACM_recov$Death_or_Transplant[ACM_recov$DEATH=="1"] <- 1
ACM_recov$Death_or_Transplant[ACM_recov$TRANSPLANT=="1"] <- 1

ACM_recov$LVEF_recovery[ACM_recov$EJECTION_FRACTION_RECOVERY==1] <- 1
ACM_recov$LVEF_recovery[is.na(ACM_recov$EJECTION_FRACTION_RECOVERY)] <- NA
ACM_recov$LVEF_recovery[ACM_recov$Death_or_Transplant==1] <- 0

ACM_recov$No_LVEF_Recovery[ACM_recov$EJECTION_FRACTION_RECOVERY==0] <- 1
ACM_recov$No_LVEF_Recovery[is.na(ACM_recov$EJECTION_FRACTION_RECOVERY)] <- NA
ACM_recov$No_LVEF_Recovery[ACM_recov$Death_or_Transplant==1] <- 0

ACM_recov_with_NA<-ACM_recov

ACM_recov<-ACM_recov[!is.na(ACM_recov$LVEF_recovery),]

ACM_recov$VarGroup <- gsub("GenotypeNegative", "GenotypeNeg_120", ACM_recov$VarGroup)

ACM_recov$TIME <- as.numeric(ACM_recov$TIME)/365

RecoveryTable <- ACM_recov %>%
  group_by(VarGroup) %>%
  summarise(
    #binary Variables
    FollowupTime_in_years=continuousVariable_summary(TIME),
    Death_or_transplant=binaryVariable_summary(Death_or_Transplant,n()),
    LVEF_recovery=binaryVariable_summary(LVEF_recovery,n()),    
    No_LVEF_Recovery=binaryVariable_summary(No_LVEF_Recovery,n())
  ) 


RecoveryTable2 <- ACM_recov %>%
  filter(VarGroup=="TTNtv" | VarGroup=="GenotypeNeg_120") %>%
  summarise(
    FollowupTime_in_years=with(.,signif(digits=2,wilcox.test(TIME ~ VarGroup)$p.val)),
    Death_or_transplant=with(., signif(digits=2,fisher.test(x=Death_or_Transplant,y=VarGroup)$p.val)),
    LVEF_recovery=with(., signif(digits=2,fisher.test(x=LVEF_recovery,y=VarGroup)$p.val)),
    No_LVEF_Recovery=with(., signif(digits=2,fisher.test(x=No_LVEF_Recovery,y=VarGroup)$p.val))
  )
#Add VarGroup column to RecoveryTable2
RecoveryTable2<-cbind(VarGroup="pvalue_TTN_GN",RecoveryTable2)


RecoveryTable<-rbind(RecoveryTable,RecoveryTable2)

RecoveryTable<-as.data.frame(t(RecoveryTable))

#Make first row column names
names(RecoveryTable) <- lapply(RecoveryTable[1, ], as.character)
RecoveryTable <- RecoveryTable[-1,] 

#Update Column names and row names
#RecoveryTable<-update_colnames_VarGroups(RecoveryTable)
#RecoveryTable<-update_rownames_characteristics(RecoveryTable)
RecoveryTable<-update_Variable_Names_for_manuscript(RecoveryTable)

#change order of columns
RecoveryTable<-RecoveryTable[,c(3,2,1,4)]


#Table 4 : Outcome based on genotype 
Outcome_wrt_genotype <- as.data.frame(RecoveryTable)









```

```{r Outcome_wrt_genotype_NAs}


RecoveryTable_NAs <- ACM_recov_with_NA %>%
  group_by(VarGroup) %>%
  summarise(
    Death_or_Transplant=sum(is.na(Death_or_Transplant)),
    LVEF_recovery=sum(is.na(LVEF_recovery)),
    No_LVEF_Recovery=sum(is.na(No_LVEF_Recovery))
  )


RecoveryTable_NAs<-as.data.frame(t(RecoveryTable_NAs))

#Make first row column names
names(RecoveryTable_NAs) <- lapply(RecoveryTable_NAs[1, ], as.character)
RecoveryTable_NAs <- RecoveryTable_NAs[-1,] 

#Update Column names and row names
#MainTable_NA<-update_colnames_VarGroups(MainTable_NA)
#RecoveryTable_NAs<-update_rownames_characteristics(RecoveryTable_NAs)
RecoveryTable_NAs<-update_Variable_Names_for_manuscript(RecoveryTable_NAs)

#change order of columns
RecoveryTable_NAs<-RecoveryTable_NAs[,c(3,2,1)]


Outcome_wrt_genotype_NAs <-as.data.frame(RecoveryTable_NAs)


```

## Table 2 : Burden analysis 
######DCM : 10VC02667 = -7+1 = two frameshifts that cancel each other out.  ie not TTNtv
######DCM : 10BK01803 has a TTNtv and a BAG3 SNP. 
######ACM : 20PC04885 has a TTNtv and a LMNA SNP. 

```{r Label VarType against SampleIDs}

RareVarRaw$Gene_VarGroup <- paste(RareVarRaw$GENE, RareVarRaw$Var_Group, sep="_")

RareVar<-RareVarRaw[,c("SampleID","Gene_VarGroup")]

RareVar <- aggregate(Gene_VarGroup ~ SampleID, data = RareVar, paste, collapse = ",")

#Merge Tables
AllSamples<-merge(RareVar, SampleIDs, by.x = "SampleID", by.y = "IID", all = TRUE) 

Gene_vector <-unique(as.vector(Gene_VarType_BurdenTest_list$V1))

for (i in 1:length(Gene_vector)) { 
  gene<-Gene_vector[i]
  AllSamples[grep(gene, AllSamples$Gene_VarGroup), gene] <- 1  

}


AllSamples[grep("TV", AllSamples$Gene_VarGroup), "ALL_GENES_except_TTN"] <- 1
AllSamples <- within(AllSamples, ALL_GENES_except_TTN[Gene_VarGroup == 'TTN_TV'] <- NA)

AllSamples[grep("TV", AllSamples$Gene_VarGroup), "ALL_GENES"] <- 1

#Manually Change One DCM annotation
AllSamples <- within(AllSamples, TTN_TV[SampleID == '10VC02667'] <- NA)


AllSamples[is.na(AllSamples)] <- 0

group_VarGroups<-as.data.frame(AllSamples[ , !names(AllSamples) %in% c("SampleID","Gene_VarGroup")])

ACM_DCM_data <- filter(group_VarGroups, group=="ACM" | group=="DCM")
ACM_HVOL_data <- filter(group_VarGroups, group=="ACM" | group=="HVOL")
DCM_HVOL_data <- filter(group_VarGroups, group=="DCM" | group=="HVOL")

```



```{r burden test}


Mean_stdDev_summary_temp <- group_VarGroups %>%
group_by(group) %>%
  summarise(
    #total=n(),
    #binary Variables
  ALL_GENES=binaryVariable_summary(ALL_GENES,n()),
  TTN_TV=binaryVariable_summary(TTN_TV,n()),
  ALL_GENES_except_TTN=binaryVariable_summary(ALL_GENES_except_TTN,n())
    ) 

Mean_stdDev_summary_temp<-as.data.frame(Mean_stdDev_summary_temp)
Mean_stdDev_summary <- data.frame(Mean_stdDev_summary_temp[,-1], row.names=Mean_stdDev_summary_temp[,1])

ACM_DCM_summary<-Summary_burden_Analysis(ACM_DCM_data,"pvalue_A_D")
ACM_HVOL_summary<-Summary_burden_Analysis(ACM_HVOL_data, "pvalue_A_H")
DCM_HVOL_summary<-Summary_burden_Analysis(DCM_HVOL_data, "pvalue_D_H")


BurdenTest_table<-t(rbind(Mean_stdDev_summary,ACM_DCM_summary, ACM_HVOL_summary, DCM_HVOL_summary))

BurdenTest_table<-update_Variable_Names_for_manuscript(BurdenTest_table)

panderOptions('table.split.table', 800)
pander(BurdenTest_table)
```

## Supp Table 7 : Summary of rare, protein altering variants in DCM-related genes detected per cohort 
```{r SuppTable_PAV_freq}


Mean_stdDev_temp <- group_VarGroups %>%
  group_by(group) %>%
    summarise_each(funs(binaryVariable_summary(.,n())), vars = BAG3_non_TV:ALL_GENES)  #BAG3_non_TV & ALL_GENES are hardcoded !!! 

Mean_stdDev_temp<-as.data.frame(Mean_stdDev_temp)
Mean_stdDev <- data.frame(Mean_stdDev_temp[,-1], row.names=Mean_stdDev_temp[,1])
colnames(Mean_stdDev)<-colnames(group_VarGroups[-1])

ExAC_freqs  = t(GeneVartype_ExAC_freqs[c("ExAC")])
ExAC_freqs_VarGroup  = t(GeneVartype_ExAC_freqs[c("Gene_VarGroup")])
colnames(ExAC_freqs) <- ExAC_freqs_VarGroup
Mean_stdDev <-merge(Mean_stdDev,ExAC_freqs,all.x=TRUE,all.y=TRUE,sort=F)

SuppTable_PAV_freq<-t(Mean_stdDev)
colnames(SuppTable_PAV_freq) <- c("ACM","DCM","HVOL","ExAC")

SuppTable_PAV_freq<-update_Variable_Names_for_manuscript(SuppTable_PAV_freq)

panderOptions('table.split.table', 800)
pander(SuppTable_PAV_freq)


#Burden Test : ACM vs ExAC
positive_ACM<-19
negative_ACM<-122
positive_ExAC<-2490
negative_ExAC<-58216
ACM_ExAC <- matrix(nrow=2, ncol=2, data=c(positive_ACM,negative_ACM,positive_ExAC,negative_ExAC))
ACM_ExAC_pvalue <- format(signif(fisher.test(ACM_ExAC)$p.value, digits=3));

```
###### Burden Test of ACM vs ExAC, p-value=`r ACM_ExAC_pvalue`




## Table 3 : Characteristics of ACM cases with and without titin truncating variants  & Outcome based on genotype 
###### The patient with 'Family history of CM' in 'Other Variants' group is the patient with BAG3tv [20WX04905]. His age at clinical assement is also 45years, much closer to TTNtv 
```{r Table 3}

#Characteristics_ACM_Plus_Outcome <-rbind(Characteristics_ACM_wrt_TTNTv,Outcome_wrt_genotype)

#Update Column names and row names
#Characteristics_ACM_Plus_Outcome<-update_colnames_VarGroups(Characteristics_ACM_Plus_Outcome)
#Characteristics_ACM_Plus_Outcome<-update_rownames_characteristics(Characteristics_ACM_Plus_Outcome)
#Characteristics_ACM_Plus_Outcome<-update_Variable_Names_for_manuscript(Characteristics_ACM_Plus_Outcome)

panderOptions('table.split.table', 800)
pander(Characteristics_ACM_wrt_TTNTv)

panderOptions('table.split.table', 800)
pander(Outcome_wrt_genotype)

```

## 

#### Table for internal use - missing data info for above table

```{r missing data info for Characteristics_ACM_Plus_Outcome table}

Characteristics_ACM_Plus_Outcome_NAs<-rbind(Characteristics_ACM_wrt_TTNTv_NAs,Outcome_wrt_genotype_NAs)

#Update Column names and row names
#Characteristics_ACM_Plus_Outcome_NAs<-update_colnames_VarGroups(Characteristics_ACM_Plus_Outcome_NAs)
#Characteristics_ACM_Plus_Outcome_NAs<-update_rownames_characteristics(Characteristics_ACM_Plus_Outcome_NAs)
Characteristics_ACM_Plus_Outcome_NAs<-update_Variable_Names_for_manuscript(Characteristics_ACM_Plus_Outcome_NAs)


panderOptions('table.split.table', 800)
pander(Characteristics_ACM_Plus_Outcome_NAs)

```

## Supplementary Table 2 : Clinical characteristics of ACM cohort based on genotype

```{r Supplementary Table 2}

panderOptions('table.split.table', 800)
pander(SuppTable_clinical_Characteristics_ACM)

```

##SUpplementary Table : List of Variants in all Cohorts
```{r VariantsList}
RareVar<-RareVarRaw

RareVar$REF_ALT <- paste(RareVar$REF, RareVar$ALT, sep=">")
RareVar$POS_REF_ALT <- paste(RareVar$POS, RareVar$REF_ALT, sep="")
RareVar$HGVSg <- paste(RareVar$CHROM, RareVar$POS_REF_ALT, sep=": g.[")
RareVar$HGVSg = paste(RareVar$HGVSg,"]",sep = "")
#chr1:g.[35227587C>G]

#RareVar$HGVSc<- gsub(x = RareVar$HGVSc,pattern = ":", replacement = ": ")
#RareVar$HGVSp<- gsub(x = RareVar$HGVSp,pattern = ":", replacement = ": ")
RareVar$HGVSc<- gsub(x = RareVar$HGVSc,pattern = ".*:", replacement = "")
RareVar$HGVSp<- gsub(x = RareVar$HGVSp,pattern = ".*:", replacement = "")

RareVar$CSQ<- gsub(x = RareVar$CSQ,pattern = "&", replacement = " & ")
RareVar$CSQ<- gsub(x = RareVar$CSQ,pattern = "_", replacement = " ")

VariantsList<-RareVar[,c("group","GENE","HGVSg","HGVSc","HGVSp","CSQ","ExAC_AF","af_filter","af_filter_pop")]

VariantsList<-update_Variable_Names_for_manuscript(VariantsList)

panderOptions('table.split.table', 800)
panderOptions('table.split.table', 8000)
pander(VariantsList)

```

